src/ripple/app/misc/HashRouter.cpp

```diff
@@ -104,7 +104,7 @@ HashRouter::shouldRelay (uint256 const& key)
     if (!s.shouldRelay(suppressionMap_.clock().now(), holdTime_))
         return boost::none;
 
-    return std::move(s.peekPeers());
+    return std::move(s.releasePeerSet());
```

Comment:
I had missed the return type.  Agreed, it won't prevent copy elision.  The move is not harmful, just unnecessary.